app/scripts/migrations/107.ts

```diff
@@ -0,0 +1,86 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+export const version = 107;
+
+interface AccountBalance {
+  address: string;
+  balance: string;
+}
+
+interface AccountTrackerControllerState {
+  accountsByChainId: Record<string, Record<string, AccountBalance>>;
+  accounts: Record<string, AccountBalance>;
+  currentBlockGasLimit: string;
+  currentBlockGasLimitByChainId: Record<string, string>;
+}
+
+/**
+ * Migrates state from the now removed CachedBalancesController to the AccountTrackerController and formats it accordingly.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  versionedData.data = transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (
+    !hasProperty(state, 'CachedBalancesController') ||
+    !isObject(state.CachedBalancesController) ||
+    !hasProperty(state.CachedBalancesController, 'cachedBalances') ||
+    !isObject(state.CachedBalancesController.cachedBalances) ||
+    !hasProperty(state, 'AccountTracker') ||
+    !isObject(state.AccountTracker)
+  ) {
+    return state;
+  }
+
+  if (!state.AccountTracker.accountsByChainId) {
+    state.AccountTracker.accountsByChainId = {};
+  }
+
+  const accountTrackerControllerState =
+    state.AccountTracker as unknown as AccountTrackerControllerState;
```

Comment:
This doesn't seem ideal but I'm not seeing a great way around it...

Reply:
dropping `accounts`, `currentBlockGasLimit`, and `currentBlockGasLimitByChainId` from `AccountTrackerControllerState` type may get you partly there. but yeah i think you're going to have to do this unless you want to loop over `accountsByChainId` and do a type assertion

Reply:
yeah I didn't actually have those originally. Its this or a type assertion / type guard